platform/mbed_watchdog_mgr.cpp

```diff
@@ -0,0 +1,112 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "mbed_watchdog_mgr.h"
+
+static bool is_watchdog_started = false; //boolean to control watchdog start and stop
+#define MS_TO_US(x) ((x) * 1000) //macro to convert millisecond to microsecond
+static uint32_t elapsed_ms = (HW_WATCHDOG_TIMEOUT / 2);
+MBED_STATIC_ASSERT((HW_WATCHDOG_TIMEOUT > 0), "Timeout must be greater than zero");
+
+/** mbed watchdog manager creates watchdog class instance with zero timeout
+ *  as mbed watchdog manager is not going to register(not going to call "start" method of Watchdog class)
+ *  its only going to call process to verify all registered users/threads in alive state
+ *
+ */
+mbed::Watchdog watchdog(0, "Platform Watchdog");
```

Comment:
why does not follow same as  `get_ticker()` below?

Reply:
Watchdog has divided into two, HW watchdog & SW watchdog,HW watchdog intern creates the instance to get access the SW watchdog methods to check states